### PR TITLE
Menu component

### PR DESCRIPTION
**Pull request by vbalan92**

### CODE REVIEW

File: elm.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
         "Hatchinq.DropDown",
         "Hatchinq.IconButton",
         "Hatchinq.List",
```
Comment (Dzhyrma) [inline elm.json:19]: You need also to bumb a version

File: src/Hatchinq/Divider.elm [hunk 2]
```diff
@@ -26,11 +30,14 @@ type alias Config =
 
 type alias InternalConfig =
     { color : Color
+    , height : Length
```
Comment (Dzhyrma) [inline src/Hatchinq/Divider.elm]: these configurations shouldn't be here. There are already a part of the Attribute

**Review by Dzhyrma — CHANGES_REQUESTED** [review]

File: src/Hatchinq/Menu.elm [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    view config
+
+
+view : Config -> List (Attribute msg) -> View msg -> Element msg
+view config attributes data =
+    let
+        elementAttributes =
+            toElement attributes
+
+        -- TODO min width?
```
Comment (vneacsu) [inline src/Hatchinq/Menu.elm]: yes, provide a min with a reasonable default

File: src/Hatchinq/Util.elm [hunk 4]
```diff
@@ -40,27 +40,60 @@ keysDownAttribute keyCodes =
...
-    Element.htmlAttribute <| Html.Events.on "keydown" (Json.Decode.andThen isKey Html.Events.keyCode)
+    Element.htmlAttribute <| Html.Events.on "keydown" (Decode.andThen isKey Html.Events.keyCode)
 
 
 takeFirstNLines : String -> Int -> String
 takeFirstNLines text numLines =
     String.join "\n" (List.take numLines (String.split "\n" text))
+
+
+onClickPropagation : Bool -> msg -> Element.Attribute msg
```
Comment (vneacsu) [inline src/Hatchinq/Util.elm]: `onClickPropagation True` reads a bit confusing. I would simply drop the parameter and name it explicitly `onClickWithoutPropagation` or something similar. This function we anyways use only like that. For all other cases, there is the plain `onClick` event.

File: src/Hatchinq/Menu.elm [hunk 5]
```diff
@@ -30,65 +29,26 @@ type alias State =
...
 
-subscriptions : Config msg -> State -> (Message msg -> msg) -> Sub msg
-subscriptions config state lift =
+subscriptions : State -> (Message msg -> msg) -> Sub msg
+subscriptions state lift =
     if state.isOpen then
-        Browser.Events.onMouseDown (outsideTarget "menu")
-            |> Sub.map lift
-
+        Browser.Events.onMouseDown (outsideTarget "menu"
```
Comment (vneacsu) [inline src/Hatchinq/Menu.elm]: indentation looks a bit funny here. You should use the `elm-format` tool which takes care of the alignment by its own.

File: src/Hatchinq/Util.elm (hunk 4 above)
Comment (dejanas) [reply src/Hatchinq/Util.elm]: I had first set it like you said but it is also used in IconButton where you can pass the propagation flag as well.

Comment (vneacsu) [reply src/Hatchinq/Util.elm]: Then I would still name it `onClickWithoutPropagation True`. sounds a bit more intuitive I would say :)

Comment (dejanas) [reply src/Hatchinq/Util.elm]: Agreed :)

**Review by vneacsu — APPROVED** [review]

**Review by vneacsu — APPROVED** [review]

**Pull request merged**